tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -248,8 +250,8 @@ static std::string DeclareLambda(const std::string &expr, const ColumnNames_t &v
 /// Resolve that alias and return the true type as string.
 static std::string RetTypeOfLambda(const std::string &lambdaName)
 {
-   auto *ti = gInterpreter->TypedefInfo_Factory((lambdaName + "_ret_t").c_str());
-   const char *type = gInterpreter->TypedefInfo_TrueName(ti);
+   const auto dt = gROOT->GetType((lambdaName + "_ret_t").c_str());
+   const auto type = dt->GetFullTypeName();
```

Comment:
```suggestion
   const auto type = dt ? dt->GetFullTypeName() : "";
```

Reply:
Do we want to return an empty string or just throw an exception? @eguiraud ?

Reply:
Indeed if the calling function(s) are assuming success then an exception might be better.

Reply:
we don't expect this to fail, ever. an assert would be most appropriate

Reply:
Assert is in!